File: BEE 1021.cpp
#include<iostream>
using namespace std;

int main(){
    double num;
    int a, b, res;
    cin >> num;
    a = num;
    cout << "NOTAS:" << endl;
    cout << a / 100 << " nota(s) de R$ 100.00" << endl;
    a = a % 100;
    cout << a / 50 << " nota(s) de R$ 50.00" << endl;
    a = a % 50;
    cout << a / 20 << " nota(s) de R$ 20.00" << endl;
    a = a % 20;
    cout << a / 10 << " nota(s) de R$ 10.00" << endl;
    a = a % 10;
    cout << a / 5 << " nota(s) de R$ 5.00" << endl;
    a = a % 5;
    cout << a / 2 << " nota(s) de R$ 2.00" << endl;
    a = a % 2;
    cout << "MOEDAS:" << endl;
    cout << a / 1 << " moeda(s) de R$ 1.00" << endl;
    b = num * 100;
    res = b % 100;
    cout << res / 50 << " moeda(s) de R$ 0.50" << endl;
    res = res % 50;
    cout << res / 25 << " moeda(s) de R$ 0.25" << endl;
    res = res % 25;
    cout << res / 10 << " moeda(s) de R$ 0.10" << endl;
    res = res % 10;
    cout << res / 5 << " moeda(s) de R$ 0.05" << endl;
    res = res % 5;
    cout << res / 1 << " moeda(s) de R$ 0.01" << endl;
    return 0;
}

File: BEE 1021.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException{
        Scanner input = new Scanner(System.in);

        double num = input.nextDouble();
        int arr[] = {10000, 5000, 2000, 1000, 500, 200, 100, 50, 25, 10, 5, 1};
        String title = "nota(s) de R$";
        int temp, n;
        n = (int)(num * 100);

        System.out.printf("NOTAS:\n");
        for (int i = 0; i < arr.length; i++) {
            temp = n / arr[i];
            n = n % arr[i];
            if (i == 6) {
                title = "moeda(s) de R$";
                System.out.printf("MOEDAS:\n");
            }
            System.out.printf("%d %s %.2f\n", temp, title, (arr[i]/100.00));
        }
    }
}